61.md

```diff
@@ -0,0 +1,145 @@
+
+| n-tag | description | expected d-tag prefixes if using kind:30161 |
+|-|-|-|
+| bookmark | uncategorized, "global" list of things a user wants to save | "i:" (kind:1 notes), "a:" (kind:30023 articles), "t:" (hashtags), "r:" (URLs) |
+| contact | pubkeys to whom the user sent DMs or started an interaction such as an one-on-one audio/video call | "p:" |
+| follow | followed pubkeys | "p:" |
+| follow-hashtag | followed hashtags | "t:" |
+| file | adds events and files (urls) to a root directory similar to "/" | "i:", "a:", "r:" (url) |
+| friend | friends' pubkeys | "p:" |
+| mute | pubkeys whose posts the user doesn't want to see in their feeds | "p" (pubkey) |
+| mute-event | muted notes | "i:" (event), "a:" (event) |
+| mute-hashtag | muted hashtags | "t:" (hashtag) |
+| mute-word | muted words | "word:" (lowercase string) |
+
+### User-generated
+
+Set names must be prefixed with a "\`" (backtick) when on `n` tags to avoid name clashing.
+To keep track of these custom names, a "Custom Set" `kind:30061` event is used for each of them.
+The `d` tag is the name. Example:
+
+```js
+{
+  kind: 30061,
+  tags: [
+    ["d", "Good Fellows"] // this name needs a ` (backtick) prefix only when set to an "n" tag
+  ]
+  // other fields
+}
+```
+
+An user-generated set may be grouped in other sets to make them related.
+The below example shows how clients can fetch `kind:30061` events with `n-tag=follow`
+to discover custom follow set names:
+
+```js
+{
+  kind: 30061,
+  tags: [
+    ["d", "Best Friends Forever"],
+    ["n", "follow"]
+  ]
+  // other fields
+}
+```
+
+For names that are secret, an encrypted `map` tag maps the obfuscated set name to its real meaning.
+Creating a private directory name example:
+
+```js
+{
+  kind: 30061,
+  tags: [
+    ["d", "14pqug65sgl"], // random and unique
+    ["n", "file"]
+    ["map", nip44Encrypt("/sub/directory")]
+  ]
+  // other fields
+}
+```
+
+## How to add items to a set
+
+There are two ways:
+
+**1) Set a `n` tag to the event to be added itself:** Do this if the event set is only made of
+events you own. For example, adding one of your long-form posts to the (obfuscated) "/sub/directory":
+
+```js
+{
+  kind: 30023,
+  tags: [
+    ["d", "<random>"],
+    ["n", "`14pqug65sgl"]
+    // other tags
+  ]
+  // other fields
+}
+```
+
+**2) Add a `n` tag to a "Set Item Reference" `kind:30382` event that references the item to be added:**
+Do this if the set may include events you don't own or other things that aren't events.
+
+The `kind:30382` event may have tags other than `n` with extra metadata about the referenced item.
+Because of that, when removing some metadata during an edit, clients should only delete the event when all tags other
+than the `d` one were removed and the content is an empty string.
+
+The `d` tag is set to the item reference with the format `<tag-referrer-name>:<tag-value>`. The notable exception
```

Comment:
Great insight! Your idea will enable removing follow-hashtag, mute-event etc.

About user-generated/custom sets, they need `kind:30061` (an alternative to same #761's kind:10008 concept, but individual events instead of a list) to keep track of their names (client can then present the custom names and the standard ones on a "dropdown" when adding something to a set). There's more to `kind:30061` need: marking a custom set name as "n=follow" will let client know the list may be used as custom feed.